spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -109,6 +110,8 @@ public InternalRow[] call(InternalRow args) {
             action = checkAndApplyStrategy(action, strategy, sortOrderString, table.schema());
           }
 
+          action = checkAndApplyBranch(table, action);
```

Comment:
Looks like only the spark WAP branch conf is considered.

Can you please also make it work with branch identifier (`branch_yourBranch`) and add testcases too? 
https://github.com/apache/iceberg/blob/master/docs/spark-writes.md#writing-to-branches

Reply:
As per this convo, rewrite data files already works with branch identifier.
https://apache-iceberg.slack.com/archives/C025PH0G1D4/p1682453453154299?thread_ts=1682025581.770009&cid=C025PH0G1D4

Reply:
not sure what you mean by make it work. how ?  it is only store procedure here, no insert statements. do you mean the table identifier reference ?

Reply:
yes. Support extracting branch info from table identifer.